scene/gui/graph_node.cpp

```diff
@@ -228,697 +243,493 @@ void GraphNode::_resort() {
 			}
 		}
 
-		if (refit_successful) { //uf refit went well, break
+		if (refit_successful) {
 			break;
 		}
 	}
 
-	/** Final pass, draw and stretch elements **/
+	// Final pass, draw and stretch elements.
 
-	int ofs = sb->get_margin(SIDE_TOP);
+	int ofs_y = sb_panel->get_margin(SIDE_TOP) + titlebar_min_size.height + sb_titlebar->get_minimum_size().height;
 
-	first = true;
-	int idx = 0;
-	cache_y.clear();
-	int w = new_size.width - sb->get_minimum_size().x;
+	slot_y_cache.clear();
+	int width = new_size.width - sb_panel->get_minimum_size().width;
 
-	for (int i = 0; i < get_child_count(); i++) {
-		Control *c = Object::cast_to<Control>(get_child(i));
-		if (!c || !c->is_visible_in_tree()) {
+	for (int i = 0; i < get_child_count(false); i++) {
+		Control *child = Object::cast_to<Control>(get_child(i, false));
+		if (!child || !child->is_visible_in_tree() || child->is_set_as_top_level()) {
 			continue;
 		}
-		if (c->is_set_as_top_level()) {
-			continue;
+
+		_MinSizeCache &msc = min_size_cache[child];
+
+		if (i > 0) {
+			ofs_y += separation;
 		}
 
-		_MinSizeCache &msc = min_size_cache[c];
+		int from_y_pos = ofs_y;
+		int to_y_pos = ofs_y + msc.final_size;
 
-		if (first) {
-			first = false;
-		} else {
-			ofs += sep;
+		// Adjust so the last one always fits perfect, compensating for numerical imprecision.
+		if (msc.will_stretch && i == children_count - 1) {
```

Comment:
This doesn't work since `children_count` only counts valid children, and `i` enumerates all children. The old `idx` variable seems to be useful after all.